storm-core/src/jvm/org/apache/storm/utils/WorkerBackpressureThread.java

```diff
@@ -43,17 +48,33 @@ static public void notifyBackpressureChecker(Object trigger) {
         }
     }
 
+    public void terminate() {
+        running = false;
+    }
+
     public void run() {
-        try {
-            while (true) {
+        while (running) {
+            try {
                 synchronized(trigger) {
                     trigger.wait(100);
                 }
                 callback.onEvent(workerData); // check all executors and update zk backpressure throttle for the worker if needed
+            } catch (InterruptedException interEx) {
+                LOG.info("WorkerBackpressureThread gets interrupted! Ignoring Exception: ", interEx);
             }
-        } catch (Exception e) {
-            throw new RuntimeException(e);
         }
     }
 }
 
+class BackpressureUncaughtExceptionHandler implements Thread.UncaughtExceptionHandler {
+    private static final Logger LOG = LoggerFactory.getLogger(BackpressureUncaughtExceptionHandler.class);
+    @Override
+    public void uncaughtException(Thread t, Throwable e) {
+        try {
+            Utils.handleUncaughtException(e);
```

Comment:
This ignores all RuntimeExceptions.  This is fine for what it was originally written for, which was around thread pools.  But in this case any RuntimeException is going to still cause the backpressure thread to exit.

@kishorvpatil would you be OK if we renamed handeUncaughtException to be more descriptive, because it is deceptive right now, and I think will cause bug while we translate code to java.  Can we call it eatRuntimeExitOnOOM.